src/Core/Services/Implementations/UserService.cs

```diff
@@ -1226,18 +1226,23 @@ public async Task<bool> CanAccessPremium(ITwoFactorProvidersUser user)
             {
                 return false;
             }
-            if (user.GetPremium())
-            {
-                return true;
-            }
-            var orgs = await _currentContext.OrganizationMembershipAsync(_organizationUserRepository, userId.Value);
-            if (!orgs.Any())
+
+            return user.GetPremium() || await this.HasPremiumFromOrganization(userId.Value);
+        }
+
+        public async Task<bool> HasPremiumFromOrganization(Guid userId)
```

Comment:
Codescene suggest (and I agree), that we should pass in a `User` object, since we already have one.

Reply:
Done - it's an `ITwoFactorProvidersUser` because that's what `CanAccessPremium` takes. We have to repeat the `user.getUserId()` check but I think that's okay, because the method is also called directly and this way it stands on its own.